src/persist-client/src/cache.rs

```diff
@@ -67,23 +77,56 @@ impl Drop for RttLatencyTask {
 
 impl PersistClientCache {
     /// Returns a new [PersistClientCache].
-    pub fn new(cfg: PersistConfig, registry: &MetricsRegistry) -> Self {
+    pub async fn new(
+        cfg: PersistConfig,
+        registry: &MetricsRegistry,
+        pubsub: Option<PersistPubSubClientConfig>,
```

Comment:
there was a really silly reason for moving the client connection into `PersistClientCache::new`, which is that we otherwise don't have access to `Metrics`. Couldn't come up with a satisfying way to make `Metrics` (and potentially `PersistConfig` for feature flag access) accessible at a higher level. We also don't _have_ to use `Metrics` and could instead use a different, more narrowly-scoped struct, but it seems much simpler to me to have everything available on one struct